### PR TITLE
Common Moduli

### PR DESCRIPTION
**Pull request by jnsiemer**
**Description**
This PR implements...
- [x] new_cyclic
- [x] new_anticyclic

, which instantiates common modulis like X^n + 1 mod q and X^n - 1 mod q more conveniently as a  `ModulusPolynomialRingZq`.

**Testing**
- [x] I added basic working examples (possibly in doc-comment)
- [x] I triggered all possible errors in my test in every possible way
- [x] I included tests for all reasonable edge cases
- [x] I provided an intuition regarding how certain inputs have to be set

**Checklist:**
- [x] I have performed a self-review of my own code
  - [x] The code provides good readability and maintainability s.t. it fulfills best practices like talking code, modularity, ...
    - [x] The chosen implementation is not more complex than it has to be
  - [x] My code should work as intended and no side effects occur (e.g. memory leaks)
  - [x] The doc comments fit our style guide

### CODE REVIEW

File: src/construction/pk_encryption/ring_lpr.rs [hunk 1]
```diff
@@ -9,11 +9,13 @@
...
+
 use super::PKEncryption;
 use qfall_math::{
     error::MathError,
     integer::{PolyOverZ, Z},
-    integer_mod_q::{Modulus, ModulusPolynomialRingZq, PolyOverZq, PolynomialRingZq, Zq},
+    integer_mod_q::{Modulus, ModulusPolynomialRingZq, PolynomialRingZq, Zq},
     rational::Q,
     traits::{Distance, GetCoefficient, Pow, SetCoefficient},
 };
```
Comment (Marvin-Beckmann) [inline src/construction/pk_encryption/ring_lpr.rs:21]: remove space

**Review by Marvin-Beckmann — APPROVED** [review]
